### ISSUE TITLE
A way to add onFinish callback?

### ISSUE THREAD

**Issue by msqar** [jcmcneal/react-step-wizard#77]
Would be nice to have an onFinish event callback passed when the last step is reached. I want to handle the finish step through the wizard's parent. Let's say i want to call backend to save all the data from the wizard, it's not good for me to do it within the last step's component. Can't seem to find a way to pass a callback prop to this wizard, is there any way?

Thanks.

**Comment by jcmcneal** [comment]
Yes, there is a way you can do it now. In your last component just compare `currentStep` === `totalSteps` and if it's true then you know you're on the last step :)

**Comment by msqar** [comment]
@jcmcneal hey, Thanks for replying :) yes, i know I could do that, but... what if I want to execute a parent's callback from that component? Just when i click on "Finish", that Finish has no nextStep() call. Because there are no other steps to go through. I just would like to execute a callback function when I click on finish, but handle it outside the step children component.

**Comment by resolritter** [comment]
Check out this implementation: https://codesandbox.io/s/react-step-wizzard-effects-suvjy?file=/src/App.js

<details>

# Code

```jsx
import React from "react";
import StepWizard from "react-step-wizard";
import "./styles.css";

const Step1 = function ({ nextStep, parentProps }) {
  return (
    <div>
      <p>Step: 1</p>
      <p>Parent props: {parentProps}</p>
      <button onClick={nextStep}>Next Step</button>
    </div>
  );
};

const Step2 = function () {
  return <span>2</span>;
};

export default function App() {
  const hasFinished = React.useRef(false);
  const onFinish = React.useRef(function () {
    if (hasFinished.current) {
      return;
    }
    hasFinished.current = true;
    console.log("Finished");
  });

  const onFinishHOC = React.useCallback(
    function (Step) {
      return function (props) {
        // do any effect you want here
        if (props.totalSteps === props.currentStep) {
          onFinish.current();
        }
        return <Step {...{ ...props }} />;
      };
    },
    [onFinish]
  );

  const steps = React.useMemo(
    function () {
      return [Step1, Step2].map((s, i) => {
        const Step = onFinishHOC(s);
        return <Step key={i} parentProps={42} />;
      });
    },
    [onFinishHOC]
  );

  return <StepWizard>{steps}</StepWizard>;
}

```
</details>

Notice how the "Finished" is only logged when you reach the final step.

> what if I want to execute a parent's callback from that component

You can host a context in the parent with a Provider and use the context in the steps.

or

You can pass the callback as props to your components. Notice how the `customValue` prop in the CodeSandbox example is available for all the steps.

**Comment by msqar** [comment]
> Check out this implementation: https://codesandbox.io/s/react-step-wizzard-effects-suvjy?file=/src/App.js
> 
> Notice how the "Finished" is only logged when you reach the final step.
> 
> > what if I want to execute a parent's callback from that component
> 
> You can host a context in the parent with a Provider and use the context in the steps.
> 
> or
> 
> You can pass the callback as props to your components. Notice how `parentProps` in the CodeSandbox example is available for all the steps.

Ohhhh my bad, didn't think about this one:

"You can pass the callback as props to your components. Notice how parentProps in the CodeSandbox example is available for all the steps."

You're absolutely right hahaha, my bad! Thanks man.
